File: src/main/java/com/github/kindrat/cassandra/client/filter/fields/EqPredicate.java
package com.github.kindrat.cassandra.client.filter.fields;

import com.datastax.driver.core.TypeCodec;
import com.github.kindrat.cassandra.client.ui.DataObject;

import java.util.Objects;

public class EqPredicate extends RowPredicate {

    public EqPredicate(String field, String value) {
        super(field, value);
    }

    @Override
    public boolean test(DataObject data) {
        TypeCodec<?> typeCodec = getColumnCodec(data.get(getField()));
        Object typedValue = typeCodec.parse(getValue());
        return Objects.equals(typedValue, data.get(getField()));
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/filter/fields/EqPredicate.kt
package com.github.kindrat.cassandra.client.filter.fields

import com.github.kindrat.cassandra.client.ui.DataObject

class EqPredicate(field: String, value: String) : RowPredicate(field, value) {

    override fun test(data: DataObject): Boolean {
        val typeCodec = getColumnCodec(data[field])
        val typedValue = typeCodec.parse(value)
        return typedValue == data[field]
    }
}